subsys/bluetooth/host/gatt.c

```diff
@@ -3090,6 +3090,23 @@ uint16_t bt_gatt_get_mtu(struct bt_conn *conn)
 	return bt_att_get_mtu(conn);
 }
 
+#if defined(CONFIG_BT_SMP)
+static bool ltk_present(struct bt_conn *conn)
+{
+	struct bt_keys *keys = conn->le.keys;
+
+	if (keys) {
+		if (conn->role == BT_HCI_ROLE_CENTRAL) {
+			return keys->keys & (BT_KEYS_LTK_P256 | BT_KEYS_PERIPH_LTK);
+		} else {
+			return keys->keys & (BT_KEYS_LTK_P256 | BT_KEYS_LTK);
+		}
+	}
+
+	return false;
+}
+#endif /* CONFIG_BT_SMP */
```

Comment:
Suggest to make this an internal function in `conn.c` so that it can be used here in GATT and modify `l2cap_check_security` to use this function too.